src/ray/core_worker/reference_count.h

```diff
@@ -44,8 +55,14 @@ class ReferenceCounter {
   /// dependencies are inlined or when the task finishes for plasma dependencies.
```

Comment:
Do you think these two cases (inlined vs. task finished) have diverged enough now that it would make sense to have separate functions?

Reply:
Yes, but I will probably address it in a follow-up PR that differentiates the 2 cases further (need to add object IDs that were nested in a return value for the task finished case).